### ISSUE TITLE
kbd input method not doing anything

### ISSUE THREAD

**Issue by hramrach** [arakiken/mlterm#59]
I tried to set input methods with mlcc , and it looks like

 - XIM works under X11 due to XIM compatibility
 - other complex methods can be configured only when the corresponding mlterm module is installed, otherwise revert to 'none'
   - they do something only when the corresponding IME service is running, otherwise do nothing

With kbd method the advised IME format is `kbd:arabic` which is recognized as valid input method and sticks in the configuration as well as any `kbd:whatever` string. However, I do not see any change to keyboard input. For example, the German layout swaps Y and Z but neither `kbd:german` nor `kbd:de` does. Not sure what `kbd:arabic` is supposed to do.

**Comment by hramrach** [comment]
Reading the code the kbd input method is limited to a few languages, and I don't know how writing is supposed to work in any of those. Then it maybe works as expected and is just not useful for me.

That leaves two minor issues
 - documenting what kbd input method is actually supposed to be doing
 - reject kbd:whatever when whatver is not a supported kbd input method

**Comment by arakiken** [comment]
Thanks.

This commit rejects unsupported kbd:whatever.
https://github.com/arakiken/mlterm/commit/88bd6e7a6a6c8544af3bd1e8a1f4f863a673582a

'kbd' input method was introduced about 20 years ago to help to input arabic, hebrew
and indic characters, but I don't think it's very useful these days.
On the other hand, as you pointed out, I think it's a useful idea to support various keyboard layouts,
so I will consider it after the next version is released.

**Comment by hramrach** [comment]
There are some situations in which no keymap may be available.

For fbdev and SDL the kernel console keymap is used. 

SDL can work without fbdev and without console. In that case there would be no place to read the keymap from. The user setting a keymap is the only way to get one then.

Have not tried running on Wayland yet. The situation there might be problematic as well.

**Comment by arakiken** [comment]
mlterm supports kbd:de, kbd:fr, kbd:es, kbd:pt and kbd:jp106 for available values of --im option.
If you specify one of these options, mlterm simply converts US keyboard to german, french, spanish or portuguese layout.
It is incomplete but I think it's better than nothing.

**Comment by hramrach** [comment]
Thanks for working on this.

As said the main problem is that it's not documented when this option can be used and for what purpose, in the mlterm documentation.

I think that running under X11 the X keymap is used, even by SDL, on console the kernel keymap is used. The question then is why this kbd option exists, what limitation it aims to overcome.

**Comment by arakiken** [comment]
I have written about the kbd input method in the manual, although it is simple,  -> https://github.com/arakiken/mlterm/commit/1b9f2e8917fa4892bb577600e4d8b6b98d766ea7#diff-8fd68581eaa74b4f314c6207857453c77dd21f20aa74b93b78646c52a715aab1R126

As I wrote before, 'kbd' input method was introduced over 20 years ago to help to input arabic, hebrew and indic characters.
I added it because it was not easy to input characters of these languages at that time, but now it's not useful in modern environments.
On the other hand, mlterm still supports old machines and OSs. (though it's just a hobby honestly)
In those environments it may be useful for a terminal emulator to have a function to remap keyboard layout even if it is ad hoc implementation.
